maintenance/wikia/deleteImageRevision.php

```diff
@@ -0,0 +1,164 @@
+		$timestamp = $this->db->selectField(
+			[ 'revision' ],
+			'rev_timestamp',
+			[
+				'rev_page' => $title->getArticleID(),
+				'rev_id' => $revisionId,
+			],
+			__METHOD__
+		);
+
+		if ( $this->isOldImageRevision( $timestamp, $title->getDBkey() ) ) {
+			$this->removeOldRevision( $title, $timestamp, $revisionId, $pageId, $reason );
+		} else {
+			$this->removeLatestRevision( $title, $timestamp, $revisionId, $pageId, $reason );
+		}
+	}
+
+	private function removeLatestRevision( Title $title, string $revisionTimestamp, int $revisionId, int $pageId, string $reason ) {
+		$this->output(__METHOD__ . "\n");
+
+		if ( !$this->hasOldRevisions( $title->getDBkey() ) ) {
+			$file = wfLocalFile( $title );
+
+			// intentionally left as false to enforce removal of whole File by FileDeleteForm::doDelete when there is only one revision
+			$oldfile = false;
+			if ( !FileDeleteForm::doDelete( $title, $file, $oldfile, $reason, false )->isOK() ) {
+				\Wikia\Logger\WikiaLogger::instance()->error(
+					"deleting file was not successful",
+					[
+						'page_id' => $pageId,
+						'revision_id' => $revisionId
+					]
+				);
+			}
+		} else {
+			$this->revertToPreviousRevision( $title, $reason );
+			wfWaitForSlaves();
+			$this->removeOldRevision( $title, $revisionTimestamp, $revisionId, $pageId, $reason );
+		}
+	}
+
+	private function revertToPreviousRevision( Title $title, string $comment ) {
+		$this->output(__METHOD__ . "\n");
+
+		$archiveName = $this->db->selectField(
+			['oldimage'],
+			'oi_archive_name',
+			[
+				'oi_name' => $title->getDBkey()
+			],
+			__METHOD__,
+			['ORDER BY oi_timestamp desc']
+		);
+
+		//TODO: check on sandbox on test wiki if this method works as expected
+		$this->output( $archiveName . "\n" );
+		$oldLocalFile = OldLocalFile::newFromArchiveName( $title, RepoGroup::singleton()->getLocalRepo(), $archiveName );
+		$source = $oldLocalFile->getArchiveVirtualUrl( $oldLocalFile->getArchiveName() );
+		$this->output( $source . "\n" );
+
+		$status = $oldLocalFile->upload( $source, $comment, $comment );
+
+		if( !$status->isGood() ) {
+			\Wikia\Logger\WikiaLogger::instance()->error("failed to revert file to previous revision" );
+		}
+	}
+
+	private function hasOldRevisions( $fileName ):bool {
+		$this->output(__METHOD__ . "\n");
+
+		return !empty(
+			$this->db->selectField(
+				[ 'oldimage' ],
+				'1 as has_old',
+				[ 'oi_name' => $fileName ]
+			)
+		);
+	}
+
+	private function removeOldRevision( Title $title, string $revisionTimestamp, int $revisionId, int $pageId, string $reason ) {
+		$this->output(__METHOD__ . "\n");
+
+		$oldLocalFile = OldLocalFile::newFromTitle( $title, RepoGroup::singleton()->getLocalRepo(), $revisionTimestamp );
+		$oldimage = $oldLocalFile->getArchiveName();
+
+		// if $oldimage is empty, FileDeleteForm::doDelete would delete whole file instead of single revision
+		if ( !empty( $oldimage ) ) {
+			// TODO: file is removed but still appears in file history with broken thumbnail
```

Comment:
![INFO](https://sonarsource.github.io/sonar-github/severity-info.png 'Severity: INFO') Complete the task associated to this "TODO" comment. [![rule](https://sonarsource.github.io/sonar-github/rule.png)](https://sonarqube.wikia-inc.com/coding_rules#rule_key=php%3AS1135)